libvirt/libvirt_test.go

```diff
@@ -157,3 +238,74 @@ func TestLibirt(t *testing.T) {
 		So(len(domains), ShouldResemble, 0)
 	})
 }
+
+func TestLibvirt(t *testing.T) {
+	Convey("Get Interface Name", t, func() {
+		conn, err := libvirt.NewVirConnection("test:///default")
+		conn.DomainCreateXMLFromFile("test_domain.xml", 0)
+		defer conn.CloseConnection()
+		So(err, ShouldBeNil)
+		name := "instance-00000031"
+		domain, err := GetInstanceByDomainName(conn, name)
+		So(err, ShouldBeNil)
+		interf, err := GetDomainInterfaces(domain)
+		So(err, ShouldBeNil)
+		So(len(interf), ShouldBeGreaterThan, 0)
+	})
+	Convey("Get Disk Name when disk don't exist", t, func() {
+		conn, err := libvirt.NewVirConnection("test:///default")
+		conn.DomainCreateXMLFromFile("test_domain.xml", 0)
+		defer conn.CloseConnection()
+		So(err, ShouldBeNil)
+		name := "instance-00000031"
+		domain, err := GetInstanceByDomainName(conn, name)
+		disk, err := GetDomainDisks(domain)
+		So(err, ShouldBeNil)
+		So(len(disk), ShouldResemble, 1)
+	})
+	Convey("Get Nova Metadata info", t, func() {
+		conn, err := libvirt.NewVirConnection("test:///default")
+		conn.DomainCreateXMLFromFile("test_domain.xml", 0)
+		defer conn.CloseConnection()
+		So(err, ShouldBeNil)
+		name := "instance-00000031"
+		domain, err := GetInstanceByDomainName(conn, name)
+		metadata, err := GetNovaMetadata(domain)
+		So(err, ShouldBeNil)
+		emptyMap := make(map[string]string)
+		So(metadata, ShouldHaveSameTypeAs, emptyMap)
+		So(len(metadata), ShouldResemble, 10)
+	})
+	Convey("ParseMemStats", t, func() {
+		Array := make([]libvirt.VirDomainMemoryStat, 3)
+		Array[0].Tag = 77
+		Array[0].Val = 7
+		Array[1].Tag = 33
+		Array[1].Val = 3
+		Array[2].Tag = 55
+		Array[2].Val = 5
+
+		var nr int32 = 33
+		parseMemStats(Array, nr)
+	})
+	Convey("ParseMemStats with empty array", t, func() {
+		emptyArray := make([]libvirt.VirDomainMemoryStat, 0)
+		var nr int32 = 33
+		parseMemStats(emptyArray, nr)
+	})
+	Convey("Get instances ID", t, func() {
+		conn, err := libvirt.NewVirConnection("test:///default")
+		defer conn.CloseConnection()
+		So(err, ShouldBeNil)
+		ids, err := GetInstanceIds(conn)
+		So(err, ShouldBeNil)
+		emptyArray := make([]string, 0)
+		So(ids, ShouldHaveSameTypeAs, emptyArray)
```

Comment:
Could you check if there are correct values of Ids? Now, you are checking only if there is correct type of returned value.